gateway-ha/src/main/java/io/trino/gateway/ha/persistence/dao/SelectorsDao.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.persistence.dao;
+
+import io.trino.gateway.ha.router.ResourceGroupsManager;
+import org.jdbi.v3.sqlobject.customizer.BindBean;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+
+import java.util.List;
+
+public interface SelectorsDao
+{
+    @SqlQuery("""
+            SELECT * FROM selectors
+            """)
+    List<Selectors> findAll();
+
+    @SqlQuery("""
+            SELECT * FROM selectors
+            WHERE resource_group_id = :resourceGroupId
+            """)
+    List<Selectors> findByResourceGroupId(long resourceGroupId);
+
+    @SqlQuery("""
+            SELECT * FROM selectors
+            WHERE
+                resource_group_id = :resourceGroupId
+                AND priority      = :priority
+                AND user_regex    IS NOT DISTINCT FROM :userRegex
```

Comment:
Is that "IS NOT DISTINCT FROM" supported by all databases (postgresql, myself, and soon oracle)

Reply:
"IS NOT DISTINCT FROM" is defined in SQL:2003 standard. Trino supports it 10 years ago. PostgreSQL and MySQL supports it. Not sure about Oracle.